pkg/driver/controllerserver.go

```diff
@@ -216,24 +220,28 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	}
 	klog.Infof("Deleting volume %v", volumeID)
 	secretMap := req.GetSecrets()
+	bucketName := secretMap["bucketName"]
+
+	if bucketName != "" {
+		klog.V(3).Infof("Not deleting user provided bucket: %v", secretMap["bucketName"])
```

Comment:
Shouldn't we exit after logging this message ?

Reply:
It can happen that user updates the secret map with bucketName before the volume(with tempBucket) gets a chance to be deleted. in this scenario, if we exit, tempBucket won't be deleted

Reply:
1. We should support auto bucket name, in case the name not provided by user.
2. Instead, the bucket should be delete as per `reclaimPolicy` set in SC. 
    Additionally, in the secret provide a field `bucketProctection = retain` to protect the bucket. 
     - if  `reclaimPolicy: Retain`  - do not delete the bucket  
     - if `reclaimPolicy: Delete` - check the `bucketProctection` filed in the secret. If not `retain` then delete the bucket.

Reply:
so if we want a bucket with rndomly generated name to be retained for user, we need to make sure user knows the name, don't we? How the user will know what is the bucket's name? they will need to check our plugin's logs 
imagine the retain policy is on and user deletes the volume, then wants to get the bucket after some time. They will have to sit through our logs unless we come up with a mechanism to persist the name of the retained bucket somewhere.
My idea is to only support auto generated bucket that are going to be deleted.
If user wants to retain the bucket, they should provide their bucket which name they already know. It won't be deleted

Reply:
if we want to retain an autogenerated bucket, we then need to suffix it with a timestamp. So user knows for sure what bucket is associated with what workload

Reply:
@minbaev Agree, the bucket name format should have time stamp

Reply:
@nkkashyap please take a look at the change i implemented